mm2src/coins/eth/v2_activation.rs

```diff
@@ -916,3 +921,38 @@ fn compress_public_key(uncompressed: H520) -> MmResult<H264, EthActivationV2Erro
     let compressed = public_key.serialize();
     Ok(H264::from(compressed))
 }
+
+#[cfg(not(target_arch = "wasm32"))]
+async fn run_db_migration_for_new_eth_pubkey(ctx: &MmArc, db_id: String) -> MmResult<(), EthActivationV2Error> {
+    info!("Public key hash: {db_id:?}");
+
+    let mut db_migration_sender = ctx
+        .db_migration_watcher
+        .as_option()
+        .expect("Db migration watcher isn't intialized yet!")
+        .get_sender();
+    db_migration_sender
+        .send(db_id)
+        .await
+        .map_to_mm(|err| EthActivationV2Error::InternalError(err.to_string()))?;
+
+    Ok(())
+}
+
+pub(super) async fn eth_shared_db_id(coin: &EthCoin, ctx: &MmArc) -> Option<String> {
+    // Use the hd_wallet_rmd160 as the db_id in HD mode since it's unique to a device and not tied to a single address
+    coin.derivation_method()
+        .hd_wallet()
+        .map(|_| ctx.default_shared_db_id().to_string())
+}
+
+pub(super) async fn eth_account_db_id(coin: &EthCoin) -> Option<String> {
+    if let Some(hd_wallet) = coin.derivation_method().hd_wallet() {
+        return hd_wallet
+            .get_enabled_address()
+            .await
+            .map(|addr| hex::encode(dhash160(addr.pubkey().as_bytes())));
```

Comment:
We should use the address without the 0x prefix as the `eth_account_db_id` which will be an `H160` pubkey hash similar to other db_ids.

Reply:
We can use same pubkey hash as utxo and tendermint for external EVM wallets btw but it will be a bit more complex to do so. We request from the user to sign a message and extract the public key from it, it's actually implemented for metamask here for uncompressed format https://github.com/KomodoPlatform/komodo-defi-framework/blob/0058f7107509aade6268c56cf3f33933d8220d3b/mm2src/crypto/src/metamask_ctx.rs#L56-L73
But we use compressed format for getting pubkey hash for utxo and tendermint so we should do the same for EVM.  I will check if I can get the same public key from metamask as the one from Keplr for the same seed using this way.

Reply:
done